src/arrayMethodReduce.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accum;
+
+    if (this.length === 0) {
+      return initialValue;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === 0) {
+        if (arguments.length === 1) {
+          accum = this[i];
+          i++;
+        } else {
+          accum = initialValue;
+        }
+      }
```

Comment:
You can avoid this nesting if else in loop  just make a couple of ternary expression before loop.
For accum, and for startIndex.
It wiil be more readable